src/cdomain/value/cdomains/unionDomain.ml

```diff
@@ -6,12 +6,24 @@ module type Arg =
 sig
   include Lattice.S
   val cast: ?torg:typ -> typ -> t -> t
+  val top_value: ?varAttr:attributes -> typ -> t
 end
 
 module type S =
 sig
   include Lattice.S
   type value
+  val get: fieldinfo -> t -> value
+  val get_field_and_value: t -> fieldinfo option * value
+  (** Return the field used in the last write of the union, and the value of that field *)
+
+  val replace: t -> fieldinfo -> value -> t
+  val map: (fieldinfo option -> value -> value) -> t -> t
+  val fold: (fieldinfo option -> value -> 'a -> 'a) -> t -> 'a -> 'a
+  val smart_leq: leq_elem: (value -> value -> bool) -> t -> t -> bool
+  val smart_join: join_elem:(value -> value -> value) -> t -> t -> t
+  val smart_widen: widen_elem:(value -> value -> value) -> t -> t -> t
```

Comment:
These `smart` operators seems very different from their smart counterparts for arrays. Maybe we should rename both to a better name?